### PR TITLE
add argocd client and repository resource

### PR DESCRIPTION
**Pull request by janwillies**
Signed-off-by: Jan Willies <jan.willies@accenture.com>

This PR adds initial support for ArgoCD repositories: 

- https://pkg.go.dev/github.com/argoproj/argo-cd/v2@v2.0.1/pkg/apis/application/v1alpha1#Repository
- https://argo-cd.readthedocs.io/en/stable/user-guide/private-repositories/

Fixes https://github.com/crossplane-contrib/provider-argocd/issues/1. Tests are still missing.

ArgoCD does not allow to connect via user/pass, instead we have to create a jwt at the ArgoCD API and use that. This is currently a manual step described [here](https://github.com/janwillies/provider-argocd/tree/repos#optional-start-a-local-argocd-server) and needs to be repeated every couple of days. We should be able to automate that.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	golang.org/x/tools v0.0.0-20201208062317-e652b2f42cc7 // indirect
+	google.golang.org/grpc v1.33.1
+	gopkg.in/alecthomas/kingpin.v2 v2.2.6
+	k8s.io/api v0.20.4
+	k8s.io/apimachinery v0.20.4
+	sigs.k8s.io/controller-runtime v0.8.3
+	sigs.k8s.io/controller-tools v0.5.0
+)
+
+replace (
```
Comment (hasheddan) [inline go.mod:21]: the joys of depending on another project that depends on k8s.io packages :sweat_smile:

File: apis/v1alpha1/providerconfig_types.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	Insecure *bool `json:"insecure,omitempty"`
+
+	// Credentials required to authenticate to this provider.
+	Credentials ProviderCredentials `json:"credentials"`
+}
+
+// ProviderCredentials required to authenticate.
+type ProviderCredentials struct {
+	// Source of the provider credentials.
+	// +kubebuilder:validation:Enum=None;Secret;InjectedIdentity;Environment;Filesystem
```
Comment (hasheddan) [inline apis/v1alpha1/providerconfig_types.go:45]: nitpick: guessing injected identity is not supported here

**Review by hasheddan — APPROVED** [review]
This is awesome @janwillies! I think we can go ahead and move forward with this so you and others have a base to work off of :+1:

File: apis/v1alpha1/providerconfig_types.go (hunk 2 above)
Comment (janwillies) [reply apis/v1alpha1/providerconfig_types.go:45]: indeed, I've removed it

**Comment by janwillies** [comment]
thanks @hasheddan! I'll open tickets for the missing parts once this is in 👍

**Pull request merged**